### DISCUSSION TITLE
How can i use  https://ip:port go start theia

### DISCUSSION THREAD

**Discussion by pingbaohua** [eclipse-theia/theia#14085] [Q&A]
I have a question. Using localhost will open Oracle Developer Tools for VS Code and create a connection, if creating a new connection with https and ip is blank, why is that?
correct:
![image](https://github.com/user-attachments/assets/ac965c25-6e41-45cc-b837-22dcb9abd46d)
error:
![image](https://github.com/user-attachments/assets/58a83942-e07c-4e55-9a3c-845f655710c7)

**Comment by msujew** [comment]
Hey @pingbaohua,

Theia uses subdomains to ensure that each webview has a unique domain (this is a security feature). Since subdomains don't work on IPs, webviews cannot be loaded when accessing Theia via a plain IP address.

See also https://github.com/eclipse-theia/theia/discussions/9912#discussioncomment-8586022.

**Reply by pingbaohua** [reply]
I read the following answer，There is a unified entry to modify the hostname ？

Theia serves the subdomain as {{guid}}.mini-browser.{{hostname}}/{{guid}}.webview.{{hostname}} (where {{hostname}} is usually something like localhost:3000). So you need to rewrite it to d517905a-21a9-5591-9981-2db6e3fd32d9.mini-browser.localhost:3000.

**Reply by msujew** [reply]
My comment in the thread was about using a reverse proxy to map the ingress, but this is not applicable in your case, as using an IP to access Theia prevents all subdomain access.

You can simply set the env variable `THEIA_WEBVIEW_EXTERNAL_ENDPOINT={{hostname}}` to circumvent this, but this is not recommended since it introduces security issues - webviews of extensions can then read the cookies of other webviews and even of your Theia app.

**Reply by pingbaohua** [reply]
When I set THEIA WEBVIEW EXTERNAL ENDPOINT={{hostname}}, I found that Oracle Developer Tools for VS Code could not open the new connection when using http access. Access resource error 404, but https can be accessed normally, what is the reason for this?
error info:
![image](https://github.com/user-attachments/assets/0aa797a3-5af0-48e9-8bbc-b1810e79ce02)
